components/automate-ui/src/app/pages/team-add-users/team-add-users.component.ts

```diff
@@ -0,0 +1,229 @@
+            this.loading = false;
+          }),
+          takeUntil(this.isDestroyed)
+        ).subscribe();
+ }
+
+  ngOnDestroy() {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
+  }
+
+  addUsers(): void {
+    this.addingUsers = true;
+    this.addUsersFailed = '';
+    const userIDs = Object.values(this.usersToAdd).map((user: User) => user.membership_id);
+    this.store.dispatch(new AddTeamUsers(<TeamUserMgmtPayload>{
+      id: this.teamId,
+      user_ids: userIDs
+    }));
+
+    const pendingAdd = new Subject<boolean>();
+    this.store.select(addUsersStatus).pipe(
+      filter(identity),
+      takeUntil(pendingAdd))
+      .subscribe((state) => {
+        if (state === EntityStatus.loadingSuccess) {
+          pendingAdd.next(true);
+          pendingAdd.complete();
+          this.addingUsers = false;
+          this.closePage();
+        }
+        if (state === EntityStatus.loadingFailure) {
+          this.store.select(addTeamUsersStatusHTTPError).pipe(
+            filter(identity),
+            takeUntil(pendingAdd)).subscribe((error: HttpErrorResponse) => {
+              if (error.message === undefined) {
+                this.addUsersFailed = 'An error occurred while attempting ' +
+                'to add users. Please try again.';
+              } else {
+                this.addUsersFailed = `Failed to add users: ${error.message}`;
+              }
+              this.addingUsers = false;
+          });
+        }
+      });
+  }
+
+  public getHeading(): string {
+    if (this.team) {
+      return `Add Users to ${this.team.name}`;
+    }
+    return 'Loading...';
+  }
+
+  public userNotFiltered(user: User): boolean {
+    return !(user.membership_id in this.mapOfUsersToFilter);
+  }
+
+  public usersNotFiltered(): User[] {
+    return lodashFilter(user => this.userNotFiltered(user), this.users);
+  }
+
+  usersToAddValues(): User[] {
+    return Object.values(this.usersToAdd);
+  }
+
+  addOrRemoveQueuedUser(checked: boolean, user: User): void {
+    if (checked) {
+      this.usersToAdd[user.id] = user;
+    } else {
+      delete this.usersToAdd[user.id];
+    }
+  }
+
+  isUserChecked(user: User): boolean {
+    return user.id in this.usersToAdd;
+  }
+
+  getConfirmBtnText() {
+    if (this.addingUsers) {
+      return (this.usersToAddValues().length < 2)
+        ? 'Adding User...'
+        : `Adding ${this.usersToAddValues().length} Users...`;
+    } else {
+      return (this.usersToAddValues().length < 2)
+        ? 'Add User'
+        : `Add ${this.usersToAddValues().length} Users`;
+    }
+  }
+
+  private getUsersForTeam(team: Team) {
+    this.team = team;
+    this.store.dispatch(new GetTeamUsers({ id: this.teamId }));
+    this.store.dispatch(new GetUsers());
+  }
+
+  closePage() {
+    this.router.navigate(['/settings', 'teams', this.teamId]);
+  }
+
+  getErrorMessage() {
+    return this.addUsersFailed.length > 0 ? this.addUsersFailed : undefined;
```

Comment:
This seems odd... at least this, perhaps:
```suggestion
    return this.addUsersFailed.length > 0 ? this.addUsersFailed : '';
```